### PR TITLE
James 1733

### PR DESCRIPTION
**Pull request by aduprat**
All commits will be squashed in one.
Splited for the readability.

### CODE REVIEW

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/quota/DefaultQuotaRootResolverTest.java [hunk 1]
```diff
@@ -73,19 +71,18 @@ public void getQuotaRootShouldThrowWhenUserContainsSeparator() throws Exception
         testee.getQuotaRoot(new MailboxPath("#private", "ben&wa", "INBOX"));
     }
 
-    @SuppressWarnings("unchecked")
     @Test
     public void retrieveAssociatedMailboxesShouldWork() throws Exception {
-        final MailboxMapper<TestId> mockedMapper = (MailboxMapper<TestId>) mock(MailboxMapper.class);
-        when(mockedFactory.getMailboxMapper(null)).thenAnswer(new Answer<MailboxMapper<TestId>>() {
+        final MailboxMapper mockedMapper = (MailboxMapper) mock(MailboxMapper.class);
```
Comment (mbaechler) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/quota/DefaultQuotaRootResolverTest.java]: not needed

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java [hunk 2]
```diff
@@ -115,7 +115,7 @@ private ResultSet getStoredACLRow() {
         return session.execute(
             select(CassandraACLTable.ACL, CassandraACLTable.VERSION)
                 .from(CassandraACLTable.TABLE_NAME)
-                .where(eq(CassandraMailboxTable.ID, mailbox.getMailboxId().asUuid()))
+                .where(eq(CassandraMailboxTable.ID, ((CassandraId) mailbox.getMailboxId()).asUuid()))
```
Comment (mbaechler) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java]: that's really late, could you cast at method beginning ?

File: mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/HBaseUtils.java [hunk 3]
```diff
@@ -203,8 +203,8 @@ public static Put metadataToPut(MailboxMessage<HBaseId> message) {
      * @param message message to get row key from
      * @return rowkey byte array that can be used with HBase API
      */
-    public static byte[] messageRowKey(MailboxMessage<HBaseId> message) {
-        return messageRowKey(message.getMailboxId(), message.getUid());
+    public static byte[] messageRowKey(MailboxMessage message) {
+        return messageRowKey((HBaseId) message.getMailboxId(), message.getUid());
```
Comment (mbaechler) [inline mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/HBaseUtils.java]: could you do casts at method beginning ?

File: mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/mail/HBaseMessageMapper.java [hunk 4]
```diff
@@ -164,10 +164,10 @@ public void endRequest() {
         return messageList;
     }
 
-    private List<MailboxMessage<HBaseId>> findMessagesInMailboxWithUID(Mailbox<HBaseId> mailbox, long messageUid, boolean flaggedForDelete) throws IOException {
-        List<MailboxMessage<HBaseId>> messageList = new ArrayList<MailboxMessage<HBaseId>>();
+    private List<MailboxMessage> findMessagesInMailboxWithUID(Mailbox mailbox, long messageUid, boolean flaggedForDelete) throws IOException {
+        List<MailboxMessage> messageList = new ArrayList<MailboxMessage>();
         HTable messages = new HTable(conf, MESSAGES_TABLE);
-        Get get = new Get(messageRowKey(mailbox.getMailboxId(), messageUid));
+        Get get = new Get(messageRowKey(((HBaseId) ((HBaseId) mailbox.getMailboxId())), messageUid));
```
Comment (mbaechler) [inline mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/mail/HBaseMessageMapper.java]: twice ?

File: mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java [hunk 5]
```diff
@@ -348,66 +348,66 @@ protected MessageMetaData save(Mailbox<JPAId> mailbox, MailboxMessage<JPAId> mes
...
-                .setParameter("idParam", mailbox.getMailboxId().getRawId()).setParameter("uidParam", uid).getResultList();
+                .setParameter("idParam", ((JPAId) mailbox.getMailboxId()).getRawId()).setParameter("uidParam", uid).getResultList();
     }
 
     @SuppressWarnings("unchecked")
-    private List<MailboxMessage<JPAId>> findDeletedMessagesInMailboxWithUID(Mailbox<JPAId> mailbox, long uid) {
+    private List<MailboxMessage> findDeletedMessagesInMailboxWithUID(Mailbox mailbox, long uid) {
         return getEntityManager().createNamedQuery("findDeletedMessagesInMailboxWithUID")
-                .setParameter("idParam", mailbox.getMailboxId().getRawId()).setParameter("uidParam", uid).setMaxResults(1)
+                .setParameter("idParam", ((JPAId) ((JPAId) mailbox.getMailboxId())).getRawId()).setParameter("uidParam", uid).setMaxResults(1)
```
Comment (mbaechler) [inline mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java]: twice ?

**Comment by mbaechler** [comment]
you can probably remove GuiceGenericType now ?

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java (hunk 2 above)
Comment (aduprat) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java]: OK

File: mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/HBaseUtils.java (hunk 3 above)
Comment (aduprat) [inline mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/HBaseUtils.java]: OK

File: mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/mail/HBaseMessageMapper.java (hunk 4 above)
Comment (aduprat) [inline mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/mail/HBaseMessageMapper.java]: OK

File: mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java (hunk 5 above)
Comment (aduprat) [inline mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java]: OK

**Comment by aduprat** [comment]
Yes i can.

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/quota/DefaultQuotaRootResolverTest.java (hunk 1 above)
Comment (aduprat) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/quota/DefaultQuotaRootResolverTest.java]: OK

**Comment by rouazana** [comment]
Quickly reviewed, but :+1:

**Comment by mbaechler** [comment]
:+1:

**Comment by chibenwa** [comment]
:+1:

**Comment by aduprat** [comment]
test this please

**Pull request merged**